File: WOT-iOS/WOT-iOS/core/WOTError.m
//
//  WOTError.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/2/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTError.h"

@implementation WOTError

+ (NSError *)loginErrorWithCode:(NSInteger)code userInfo:(NSDictionary *)userInfo {
    
    return [NSError errorWithDomain:@"WOTLOGIN" code:code userInfo:userInfo];

}

+ (NSError *)coreDataErrorWithCode:(NSInteger)code userInfo:(NSDictionary *)userInfo {
    
    return [NSError errorWithDomain:@"WOTCOREDATA" code:code userInfo:userInfo];
}

@end

File: WOT-iOS/WOTPivot/WOTPivot/core/errors/WOTError.swift
//
//  WOTError.swift
//  WOTPivot
//
//  Created by Pavel Yeshchyk on 4/26/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

public protocol DebugDescriptable {
    var debugDescription: String { get }
}

public protocol WOTError: Error, DebugDescriptable {
    var code: Int? { get }
    var message: String? { get }
    init(code: Int?, message: String?)
}

extension Error {
    public var debugDescription: String {
        return ".\(String(describing: self)) (code \((self as NSError).code))"
    }
}
